### PR TITLE
arm64: dts: rockchip: add support for the Orange Pi 5 Pro

### PR DESCRIPTION
**Pull request by Joshua-Riek**
Introduce a device tree for the new [Orange Pi 5 Pro](http://www.orangepi.org/html/hardWare/computerAndMicrocontrollers/details/Orange-Pi-5-Pro.html).

Build system PR: https://github.com/armbian/build/pull/6506

Cherry Picked from: https://github.com/orangepi-xunlong/linux-orangepi/commit/d910e311b5c923c7cdd85b46fc5965f4a555489e

[AR-2144]

[AR-2144]: https://armbian.atlassian.net/browse/AR-2144?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by efectn** [comment]
How similar are 5 and 5 pro devicetrees? Perhaps it would be cleaner if we include 5 devicetree and make the needed adjustments

**Comment by Joshua-Riek** [comment]
Let me take a closer look at the device trees, if we can just include the opi5 device tree that would be great.

**Comment by Joshua-Riek** [comment]
Hmm it does not look simple and would likely require some time digging through the whole OPI5 Pro device tree since what we have for the OPI5 is quite different.

For reference - Orange Pi 5 DTS are split into the following files:
https://github.com/armbian/linux-rockchip/blob/rk-5.10-rkr6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi.dtsi
https://github.com/armbian/linux-rockchip/blob/rk-5.10-rkr6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dtsi
https://github.com/armbian/linux-rockchip/blob/rk-5.10-rkr6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts

**Comment by efectn** [comment]
> Hmm it does not look simple and would likely require some time digging through the whole OPI5 Pro device tree since what we have for the OPI5 is quite different.
> 
> For reference - Orange Pi 5 DTS are split into the following files: https://github.com/armbian/linux-rockchip/blob/rk-5.10-rkr6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi.dtsi https://github.com/armbian/linux-rockchip/blob/rk-5.10-rkr6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dtsi https://github.com/armbian/linux-rockchip/blob/rk-5.10-rkr6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts

I don't really like current style. Are there any good reason that explains why Xunlong splits them into 3 files?

**Comment by Joshua-Riek** [comment]
> > Hmm it does not look simple and would likely require some time digging through the whole OPI5 Pro device tree since what we have for the OPI5 is quite different.
> > For reference - Orange Pi 5 DTS are split into the following files: https://github.com/armbian/linux-rockchip/blob/rk-5.10-rkr6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi.dtsi https://github.com/armbian/linux-rockchip/blob/rk-5.10-rkr6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dtsi https://github.com/armbian/linux-rockchip/blob/rk-5.10-rkr6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts
> 
> I don't really like current style. Are there any good reason that explains why Xunlong splits them into 3 files?

I do not like it either and have found no explanation as to why Xunlong splits them into 3 files. I think the OPI5 device tree should be re-written, but it's not something I have the time for right now.

**Comment by efectn** [comment]
> > > Hmm it does not look simple and would likely require some time digging through the whole OPI5 Pro device tree since what we have for the OPI5 is quite different.
> > > For reference - Orange Pi 5 DTS are split into the following files: https://github.com/armbian/linux-rockchip/blob/rk-5.10-rkr6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi.dtsi https://github.com/armbian/linux-rockchip/blob/rk-5.10-rkr6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dtsi https://github.com/armbian/linux-rockchip/blob/rk-5.10-rkr6/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dts
> > 
> > 
> > I don't really like current style. Are there any good reason that explains why Xunlong splits them into 3 files?
> 
> I do not like it either and have found no explanation as to why Xunlong splits them into 3 files. I think the OPI5 device tree should be re-written, but it's not something I have the time for right now.

Ok then no problem. I can take a look to refactor opi5 series devicetrees later

File: arch/arm64/boot/dts/rockchip/overlay/orangepi-5-pro-lcd.dts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+	fragment@0 {
+		target = <&dsi1>;
+		__overlay__ {
+			status = "okay";
```
Comment (efectn) [inline arch/arm64/boot/dts/rockchip/overlay/orangepi-5-pro-lcd.dts:60]: Can you move dsi properties to here

File: arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5-pro.dts [hunk 2]
```diff
@@ -0,0 +1,1052 @@
...
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-boot-on;
+		regulator-always-on;
+		enable-active-high;
+		gpios = <&gpio3 RK_PB7 GPIO_ACTIVE_LOW>;
+		vin-supply = <&vcc_3v3_s3>;
+	};
+
+        sdio_pwrseq: sdio-pwrseq {
```
Comment (efectn) [inline arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5-pro.dts]: can you fix the indentation

**Comment by igorpecovnik** [comment]
can we sqeeze this into upcoming release?

**Comment by Joshua-Riek** [comment]
Sorry for delay, I'll fix this shortly, honestly forgot :P

**Comment by igorpecovnik** [comment]
We are all crazy busy. no worries. Thanks!

File: arch/arm64/boot/dts/rockchip/overlay/orangepi-5-pro-lcd.dts (hunk 1 above)
Comment (Joshua-Riek) [reply arch/arm64/boot/dts/rockchip/overlay/orangepi-5-pro-lcd.dts:60]: Moved all dsi props to an overlay https://github.com/armbian/linux-rockchip/pull/172/commits/2c73cb81f74b30749f57f2fafcc31a8387653510.

File: arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5-pro.dts (hunk 2 above)
Comment (Joshua-Riek) [reply arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5-pro.dts]: Indentation has been fixed https://github.com/armbian/linux-rockchip/pull/172/commits/63846c517fb4beda12bffa228ff390ecb44e862f.

**Comment by igorpecovnik** [comment]
Failing :(

**Comment by Joshua-Riek** [comment]
Oh, there was a typo in the LCD overlay, thanks for catching this in https://github.com/armbian/linux-rockchip/pull/179!

**Pull request merged**